lib/beacon/config.ex

```diff
@@ -63,7 +117,35 @@ defmodule Beacon.Config do
             css_compiler: Beacon.TailwindCompiler,
             tailwind_config: Path.join(Application.app_dir(:beacon, "priv"), "tailwind.config.js.eex"),
             live_socket_path: "/live",
-            safe_code_check: false
+            # TODO: change safe_code_check to true when it's ready to parse complex codes
+            safe_code_check: false,
+            template_formats: [
```

Comment:
This API can be considered done but I'm open to suggestions. Initially I thought about a API to allow users adding or removing steps from HEEx or Markdown formats so they could add custom steps to transform them. But that proved to be more complex than just letting the user define the entire lifecycle (all steps to `load` and `render` a template in a given format) with the possibility of reusing steps like I've done for Markdown that's reusing HEEx steps.

Note that also opens the possibility to implement other engines quite easily, for eg Liquid which is a simple and safe template language. Even thought that's not the objective, I believe it's nice to have since there's no need to write extra code to support that.